src/AutoUI/Lenses/types/table.tsx

```diff
@@ -47,6 +55,27 @@ export const table: LensTemplate = {
 					sortingStateRestorationKey={`${autouiContext.resource}__sort`}
 					tagField={autouiContext.tagField}
 					enableCustomColumns
+					saveColumnPreferences={(newCols) => {
+						const columnStateRestorationKey = `${autouiContext.resource}__columns`;
+
+						const savePayload = newCols.map((c) =>
+							pick(c, TableColumnStateStoredProps),
+						);
+
+						setToLocalStorage(columnStateRestorationKey, savePayload);
+
+						const columnsAnalyticsObject = (
+							newCols as Array<TableColumn<any>>
+						).reduce((obj, item) => {
+							return { ...obj, [item.field]: item.selected };
+						}, {});
```

Comment:
Not sure if the performance would be better with this, I only bring this up because I keep hearing that `reduce` performance is not great. Go for whichever seems better
```suggestion
						const columnsAnalyticsObject = Object.fromEntries(newCols.map((col) => [col.field, col.selected]));
```